### DISCUSSION TITLE
Setter & getter separation (`useQueryValue` & `useSetQuery`)

### DISCUSSION THREAD

**Discussion by lamualfa** [47ng/nuqs#494] [Ideas]
This _setter_ & _getter_ separation concept is inspired by [`useAtomValue`](https://jotai.org/docs/core/use-atom#useatomvalue) & [`useSetAtom`](https://jotai.org/docs/core/use-atom#usesetatom) from `jotai` (_state management library_). This separation will add more granular and semantical support for the user.

**Functions**

- `useQueryValue`
- `useSetQuery`
- `useQueryValues`
- `useSetQueries`

**Use Cases**

- Component that only uses the query value without changing it (`useQueryValue` & `useQueryValues`)
  - To determine which tag is active on the article page
  - To generate a share URL when the share button is clicked
- Component that only changes the query value without using it (`useQueryValues` & `useSetQueries`)
  - To set the query values after submitting a form
  - To reset the query values when the reset button is clicked

<hr>

`useQueryState` & `useQueryStates` are still exist and used on the component that needs to read & write the query at the same time likes the search filter.

<hr>

Thanks 🌟.

**Comment by franky47** [comment]
You can do this trivially with the existing API:

```ts
// Read-only
const [value] = useQueryState('key')

// Write-only
const [, setValue] = useQueryState('key')
```

Since all hooks with the same key are synced together, you can separate read and write without any issues. The state updater function is also referentially stable across renders.

If you really want to abstract those into functions:
```ts
function useQueryValue() {
  return useQueryState('key')[0]
}

function useSetQuery() {
  return useQueryState('key')[1]
}
```

**Reply by Thiago-cpu** [reply]
But if you are just setting the state it would rerender no matter if you are using the state or not, right ?
